lib/facter/mysqld_version.rb

```diff
@@ -0,0 +1,5 @@
+Facter.add("mysqld_version") do
+  setcode do
+    Facter::Util::Resolution.exec('mysqld -V')
```

Comment:
I think the `.compact` is superfluous.  It'd remove nil elements from a list.  In the provider, the call to `.compact` is run against an anonymous, single-item list, where it will have no effect.